presto-main/src/test/java/com/facebook/presto/memory/TestQueryContext.java

```diff
@@ -132,4 +148,67 @@ public void testLegacyQueryContext()
             assertEquals(generalPool.getReservedBytes(), 0);
         }
     }
+
+    @Test(dataProvider = "testMoveTaggedAllocationsOptions")
+    public void testMoveTaggedAllocations(boolean useLegacyQueryContext)
```

Comment:
Just a comment.  When I have this situation with a boolean, I typically just have two tests that call the same internal method. I personally find it easier to read... I like the `dataProvider` stuff for more complex cases.